Source/santad/SNTApplicationCoreMetrics.m

```diff
@@ -85,9 +85,24 @@ static void RegisterMemoryAndCPUMetrics(SNTMetricSet *metricSet) {
 }
 
 static void RegisterHostnameAndUsernameLabels(SNTMetricSet *metricSet) {
-  [metricSet addRootLabel:@"host_name" value:[NSProcessInfo processInfo].hostName];
+  NSString *hostname = [NSProcessInfo processInfo].hostName;
+
+  [metricSet addRootLabel:@"host_name" value:hostname];
   [metricSet addRootLabel:@"username" value:NSUserName()];
   [metricSet addRootLabel:@"job_name" value:@"santad"];
+  [metricSet addRootLabel:@"service_name" value:@"santa"];
+
+  // get extra root labels from configuration
+  SNTConfigurator *config = [SNTConfigurator configurator];
+
+  NSDictionary *extraLabels = [config extraMetricLabels];
+
+  if ([extraLabels count] >= 0) {
+    NSLog(@"Extra labels %@", extraLabels);
+    for (NSString *key in extraLabels) {
+      [metricSet addRootLabel:key value:(NSString *)extraLabels[key]];
```

Comment:
Will this do-the-right-thing if the config contains a label already declared, like host_name? What happens if the "value" is empty? Can that be used to un-set a default label?